src/linelist.jl

```diff
@@ -1,132 +1,229 @@
+function Base.show(io::IO, m::MIME"text/plain", b::Formula)
+    print(io, *(b.atoms...))
+end
+
+function (==)(b1::Formula, b2::Formula)
+    b1.atoms == b2.atoms
+end
+
+function hash(b::Formula, h::UInt)
+    hash(b.atoms, h)
 end
 
 """
-    strip_ionization(code)
+    ismolecule(b::Formula)
 
-get the chemical symbol for the element of the species
+`true` when `b` is composed of more than one atom
 """
-strip_ionization(code::AbstractString)::String = split(code, '_')[1]
+ismolecule(b::Formula) = length(b.atoms) > 1
 
 """
-    ismolecule(species)
+    mass(b::Formula)
 
-true if the string passed represents a molecule (with or without its ionization state)
+Returns the mass [g] of `b`.
 """
-function ismolecule(species)
-    count = 0
-    for c in species
-        if c == '_'
-            break
-        else
-            count += isdigit(c) + isuppercase(c)
-        end
-    end
-    count > 1
+function mass(b::Formula)
+    sum(atomic_masses[a] for a in b.atoms)
 end
 
+const roman_numerals = ["I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X"]
+"""
+Represents an atom or molecule (a `Formula`) with a particular number of electrons (regardless of 
+their configuration).
 """
-    get_atoms(molecule)
+struct Species
+    formula::Formula
+    charge::Int
+end
 
-Get the atoms that make up a diatomic molecule
 """
-function get_atoms(molecule)
-    if '_' in molecule
-        molecule = split(molecule,'_')[1]
+    Species(code::AbstractString)
+
+Parse the "species code" as it is often specified in linelists and return a the "astronomy" 
+notation. 01.00 → "H_I", 02.01 → "He_II", 02.1000 → "He_II", 0608 → "CO_I", etc.  
+"""
+function Species(code::AbstractString)
+    code = strip(code, ['0', ' '])
+    toks = split(code, [' ', '.', '_'])
+    if length(toks) > 2
+        throw(ArgumentError(code * " isn't a valid species code"))
     end
-    if molecule[end] == '2'
-        el1 = molecule[1:end-1]
-        el2 = molecule[1:end-1]
-    elseif isuppercase(molecule[2])
-        el1, el2 = molecule[1:1], molecule[2:end]
-    elseif isuppercase(molecule[3])
-        el1, el2 = molecule[1:2], molecule[3:end]
+    formula = Formula(toks[1])
+    charge = if length(toks) == 1 || length(toks[2]) == 0
+        0 #no charge specified -> assume neutral
     else
-        throw(ArgumentError("This doesn't look like a diatomic molecule: $(molecule)"))
+        charge = findfirst(toks[2] .== roman_numerals)
+        charge = (charge isa Int ? charge : parse(Int, toks[2]))
+        #if this is a MOOG-style numeric code, the charge is correct, otherwise subtract 1
+        if tryparse(Float64, code) == nothing 
+            charge -= 1
+        end
+        charge
     end
-    el1, el2        
+    Species(formula, charge)
+end
+
+#useful constants
+const H_I = Species("H_I")
+const H_II = Species("H_II")
+const He_I = Species("He_I")
+const He_II = Species("He_II")
+const He_III = Species("He_III")
```

Comment:
As we previously discussed, I think it would be a lot better if these could be placed in a submodule of some kind so that these constants are just floating in the global namespace.

Reply:
I think a submodule consisting only of `Formula`, `Species`, these, and well-known (i.e. not passable by the user) atomic data would make sense.  It could be called `Chemistry` or `Particles`?

Reply:
Makes sense to me

Reply:
I started to do this, but I'm not convinced anymore.  I think the clunkiness of julia's include/import system relative to python combined with how multiple dispatch makes the code much harder to follow.  For the continuum opacities, which are used in exactly one place, it nicely separates functionality and namespaces, but this code is so integrated with everything else, I don't think it makes clearer to have a bunch of qualified imports.  I'm still open to doing this in the future, but I think it would make more sense to refactor the whole package into submodules rather than doing it piecemeal.

Reply:
As an alternative solution to this problem, what about placing these in a (`const`) named tuple?